web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx

```diff
@@ -61,14 +62,45 @@ export function SetupConnect(
     username,
     path: Path.ConnectMyComputer,
   });
+  const [showHint, setShowHint] = useState(false);
 
   const { node, isPolling } = usePollForConnectMyComputerNode({
     username,
     clusterId,
-    pingInterval,
+    // If reloadUser is set to true, the polling callback takes longer to finish so let's increase
+    // the polling interval as well.
+    pingInterval: showHint ? pingInterval * 2 : pingInterval,
+    // Completing the Connect My Computer setup in Connect causes the user to gain a new role. That
+    // role grants access to nodes labeled with `teleport.dev/connect-my-computer/owner:
+    // <current-username>`.
+    //
+    // In certain cases, that role might be the only role which grants the user the visibility of
+    // the Connect My Computer node. For example, if the user doesn't have a role like the built-in
+    // access which gives blanket access to all nodes, the user won't be able to see the node until
+    // they have the Connect My Computer role in their cert.
+    //
+    // As such, if we don't reload the cert during polling, it might never see the node. So let's
+    // flip it to true after a timeout.
+    reloadUser: showHint,
   });
 
-  const [showHint, setShowHint] = useState(false);
+  const { agentMeta, updateAgentMeta, nextStep } = props;
+  const handleNextStep = useCallback(() => {
+    if (!node) {
+      return;
+    }
+
+    updateAgentMeta({
+      ...agentMeta,
+      // Node is an oddity in that the hostname is the more
+      // user identifiable resource name and what user expects
+      // as the resource name.
+      resourceName: node.hostname,
+      node,
+    });
+    nextStep();
+  }, [node, updateAgentMeta, agentMeta, nextStep]);
```

Comment:
I did it mostly out of habit. Technically this makes sure that `ActionButtons` is not going to receive new props on every render, as otherwise each render of `SetupConnect` would create a function with new identity and then pass it to `ActionButtons`.

I assumed that passing the same function to `ActionButtons` would avoid rerendering, but that's not how React works. Whenever the state of a component changes, React rerenders that component and all of its children. To make the component not rerender, we'd need to use `useMemo` or something similar.

Thanks for catching me on this, I was so used to how languages/frameworks such as Elm work that I forgot its not the case for React.